services/spoonacular/test/test_service.py

```diff
@@ -0,0 +1,77 @@
+        remote = SpoonacularJSONRemote()
+    else:
+        print("Mocking remotes...")
+        remote = SpoonacularMOCKRemote()
+    spoonacular_service = SpoonacularService(remote, 'pork')
+
+    def test_apiKey(self):
+        self.assertEqual(self.remote.api_token, "0a3a6b562932438aaab0cb05460096de")
+    
+    def test_get_sourceURL(self):
+        self.assertEqual(self.spoonacular_service.get_sourceURL(), "http://www.food.com/recipe/pork-piccata-224998")
+    
+    def test_get_summary(self):
+        self.assertEqual(self.spoonacular_service.get_summary(), "Pork Piccatan is a <b>dairy free</b> main course. This recipe serves 4. For <b>$1.76 per serving</b>, this recipe <b>covers 19%</b> of your daily requirements of vitamins and minerals. One serving contains <b>251 calories</b>, <b>26g of protein</b>, and <b>12g of fat</b>. 2 people have made this recipe and would make it again. A mixture of pepper, center cut pork chops, chicken broth, and a handful of other ingredients are all it takes to make this recipe so tasty. To use up the lemon rind you could follow this main course with the <a href=\"https://spoonacular.com/recipes/lemon-almond-cake-with-lemon-curd-filling-198923\">Lemon-Almond Cake with Lemon Curd Filling</a> as a dessert. From preparation to the plate, this recipe takes approximately <b>17 minutes</b>. All things considered, we decided this recipe <b>deserves a spoonacular score of 68%</b>. This score is good. Similar recipes include <a href=\"https://spoonacular.com/recipes/pork-piccata-my-way-111393\">Pork Piccata My Way</a>, <a href=\"https://spoonacular.com/recipes/pork-piccata-111568\">Pork Piccata</a>, and <a href=\"https://spoonacular.com/recipes/pork-piccata-708193\">Pork Piccata</a>.")
+       
+    def test_get_ingredients(self):
+        self.assertEqual(self.spoonacular_service.get_ingredients(), ['black pepper 0.125 teaspoon', 'capers 2.0 teaspoons', 'chicken broth 1.0 cup', 'fresh parsley 3.0 tablespoons', 'italian seasoned breadcrumbs 0.3333333333333333 cup', 'lemon juice 1.0 tablespoon', 'lemon rind 1.0 teaspoon', 'olive oil 1.0 tablespoon', 'pork chops 16.0 ounce', 'shallot 0.25 cup'])
+    
+    def test_get_healtScore(self):
+        self.assertEqual(self.spoonacular_service.get_healthScore(), 17.0)
+    
+    def test_get_title(self):
+        self.assertEqual(self.spoonacular_service.get_title(), "Pork Piccata")
+    
+    def test_get_cookingTime(self):
+        self.assertEqual(self.spoonacular_service.get_cookingTime(), 17)
+    
+    def test_get_vegetarian(self):
+        self.assertEqual(self.spoonacular_service.get_vegetarian(), False)
+    
+    def test_get_vegan(self):
+        self.assertEqual(self.spoonacular_service.get_vegan(), False)
+
+    def test_get_search_options(self):
+        search_options = self.remote.get_search_options('pork')
+        self.assertEqual(search_options, "includeIngredients=pork&diet=omnivore&maxReadyTime=90&apiKey=0a3a6b562932438aaab0cb05460096de")
```

Comment:
apiKey in preferences?

Reply:
yes, so changing it is easier and it is possible to change to a other pay-plan

Reply:
Why is it in the URL here then?

`apiKey=0a3a6b562932438aaab0cb05460096de"

Reply:
That's only to check if the import of the api-key and the other search options are set correct.

Should I get the key there also from the prefs ? or should I insert the api-key into the code itself?

Reply:
Changing the apiKey shouldn't make a test fail, I think

Reply:
Yeah that's right, but how should I fix it?

Getting the assertEquals also from pref or "hardcoding" the api_key into the code

Reply:
For now we have to ways to test – the real remotes and the mocked remotes. Both of them should pass. Instead of having two separate runs we could combine them.

Instead of:
```
if 'DONOTMOCK' in environ:
    remote = RealRemote()
else:
   remote = MockRemote()

service = Service(remote)
```

We could run the tests two times using a `TestLoader`. I'll look into it. Until then don't worry about 100% coverage.

Reply:
Great keep mi posted